pipeline/flows/matching.py

```diff
@@ -115,6 +116,8 @@ def matching_flow(defaults: Dict = {}) -> Flow:
             users_w_profile, recent_matches, output_matches
         )
 
+        display_internal_matching_metrics.run(users_w_data, recent_matches)
```

Comment:
Ah, I probably confused you with my note.

If we are running a Prefect task in normal Python code, then we use `task.run()`.

BUT, when a Prefect task runs in a flow context (which is what is happening in this file), you can use it as if it were a function, and Prefect will automatically handle it.

On line 56 of this file, we have:

```python
with Flow(name="matching_flow") as flow:
```

And all the code that is intended under that is part of the flow context.

So instead, you can change your line to:

```python
display_internal_matching_metrics(users_w_data, ranked_matches)
```

Also, don't forget that you need to pass in ranked_matches, not recent matches.

I would recommend changing this to just use the matching output as an argument instead of the individual variables. But now that I look at the code, I realize that `compute_matches()` only returns the output matches, not the full output.

To get around that, try making the following changes:

```python
# On line 39 of the matching flow code, return both the output and the matches:
@prefect.task
def compute_matches(inp: MatchingInput, engine_id: EngineId) -> List[Match]:
    ...
    return output, output.matches

# On line 113 of the matching flow code, I'm not sure if this will work, but try this and let me know what happens:
matching_output, output_matches = compute_matches(inp, param_engine)

# Then, on line 119 of the flow code, you can do this:
display_internal_matching_metrics(matching_output)

# And finally, update your display task in the load module:
@task
def display_internal_matching_metrics(output: MatchingOutput):
  users = output.users
  ranked_matches = output.internal_data.ranked_matches
  ...
```

The `run flow matching` to see if it works.